### PR TITLE
Solution for missing logo in printable versions of reports issue #962

### PR DESCRIPTION
**Pull request by 2208Abhinav**
Logo was not previewed in the printable version of report

Before
35188139-c8653818-fe30-11e7-8a07-44d9b95e0064

Now
screenshot from 2018-01-21 18-15-38

As you can see I have reduced the size of image to 56px by 45px because without this the result was

screenshot from 2018-01-21 21-06-29

### CODE REVIEW

**Comment by alpinus4** [comment]
In my previous comment I meant keeping the proportions @2208Abhinav
It's important

**Comment by 2208Abhinav** [comment]
I get it you want that the ratio of dimensions should relate original image

**Comment by 2208Abhinav** [comment]
Please review the changes I have changed to 90.6 X 56
Original Size 2913 X 1800
screenshot from 2018-01-21 21-18-11

**Comment by 2208Abhinav** [comment]
@alpinus4 I didnt't get your point could you please elaborate.

**Comment by alpinus4** [comment]
@teryhill @aethelwulffe Could you take a look? Is there necessary to cange commits messages?

@2208Abhinav In the future don't use commit messages like update xyz, update xyz, .....
Such messages don't give too much information. Won't be better when you use for ex. _Changed image size in xyz.php_ or _Corrected image path_ or anything similar??

**Comment by 2208Abhinav** [comment]
Thank you buddy now I understand

File: interface/patient_file/report/custom_report.php [hunk 1]
```diff
@@ -484,12 +484,7 @@ function next_prev(action){
   echo '<page_header style="text-align:right;" class="custom-tag"> ' . xlt("PATIENT") . ':' . text($titleres['lname']) . ', ' . text($titleres['fname']) . ' - ' . $titleres['DOB_TS'] . '</page_header>    ';
   echo '<page_footer style="text-align:right;" class="custom-tag">' . xlt('Generated on') . ' ' . oeFormatShortDate() . ' - ' . text($facility['name']) . ' ' . text($facility['phone']) . '</page_footer>';
 
-  // Use logo if it exists as 'practice_logo.png' in the site dir
-  // old code used the global custom dir which is no longer a valid
-   $practice_logo = "$OE_SITE_DIR/images/practice_logo.png";
-   if (file_exists($practice_logo)) {
-        echo "<img src='$practice_logo' align='left'><br />\n";
-     } 
+  echo "<img src='../../../sites/default/images/practice_logo.png' align='left' style='width : 56px ; height : 90.6px'><br>";
```
Comment (aethelwulffe) [inline interface/patient_file/report/custom_report.php]: At the beginning of this file, the `require_once("../../globals.php");` allows you to use the variable $OE_SITE_DIR
Without this variable, if you have more than one /site directory, this code will be broken for any site other than /sites/default.
  Revise this to use the current site session variable above.

Comment (2208Abhinav) [reply interface/patient_file/report/custom_report.php]: okay I will try to implement that, thanks for review.

**Comment by aethelwulffe** [comment]
Next issue with this:
Most clinics will not want the Librehealth logo plastered on the reports.
This code might be best simply taken out, or use a conditional statement that renders the logo if the configuration includes a clinic-specific letterhead.  More to the point, If the feature is included at all, it should have a toggle, and/or specifically have an HTML letterhead file that it uses/includes in these cases.  The letterhead file should be editable within the application, and be /sites directory specific.  There is even a case for making it specific to the facility/clinic combination.

**Comment by 2208Abhinav** [comment]
You want to include two options 
First clinic's logo
If that not present than Librehealth logo

**Comment by teryhill** [comment]
The current code looks for the practice_logo if it exists it is used. The user can put what ever logo they want there. I don't think any code changes are needed unless you want to make one to use fidderent file types, jpg,gif,png etc.

**Comment by 2208Abhinav** [comment]
So now you want me to just use $OE_SITE_DIR variable in the image path and provide no other options?

**Comment by 2208Abhinav** [comment]
@teryhill I also agree with the point that user can manually put the required logo in the directory.

**Comment by teryhill** [comment]
@2208Abhinav  look into allowing the user to put any type of image file in the directory. like they have a practice_logo.gif, or practice _logo.png or practice_logo.jpeg. something like this:

$practicelogo = preg_grep('~practice_logo\.(gif|png|jpg|jpeg)$~i', $practicelogo);

**Comment by 2208Abhinav** [comment]
@teryhill please review the changes
I have used the `explode()` function to get the absolute directory of the image while using the `$OE_SITE_DIR` variable as mentioned by @aethelwulffe.
I did this because in the actual code the image path was rendered as **http://localhost/LibreEHR/interface/patient_file/report/sites/default/images/practice_logo.png** that's why the image was not shown and to get the correct path **http://localhost/LibreEHR/sites/default/images/practice_logo.png**.

I have allowed the facility to use any extensions use conditional statements.

**Comment by aethelwulffe** [comment]
Any of the above is fine.  Something in documentation or an obvious tool in the globals config would be better, but this is fine.
The practice logo != login logo, then we are much better off.  I still think that a custom letterhead tool would be a nice feature.

File: interface/patient_file/report/custom_report.php [hunk 2]
```diff
@@ -484,12 +484,17 @@ function next_prev(action){
...
 
-  // Use logo if it exists as 'practice_logo.png' in the site dir
-  // old code used the global custom dir which is no longer a valid
-   $practice_logo = "$OE_SITE_DIR/images/practice_logo.png";
-   if (file_exists($practice_logo)) {
-        echo "<img src='$practice_logo' align='left'><br />\n";
-     } 
+  //use logo if it exists in directory
+  if (file_exists("$OE_SITE_DIR/images/practice_logo.png"))  $practice_logo = "$OE_SITE_DIR/images/practice_logo.png";
+  else if (file_exists("$OE_SITE_DIR/images/practice_logo.gif"))  $practice_logo = "$OE_SITE_DIR/images/practice_logo.gif";
```
Comment (teryhill) [inline interface/patient_file/report/custom_report.php]: Would it be easier to add a global variable and let the user specify the name for the practice logo, then they could specify practice_logo.gif or practice_logo.tiff etc

File: interface/patient_file/report/custom_report.php [hunk 3]
```diff
@@ -484,12 +484,17 @@ function next_prev(action){
...
+  else if (file_exists("$OE_SITE_DIR/images/practice_logo.jpg"))  $practice_logo = "$OE_SITE_DIR/images/practice_logo.jpg";
+  else if (file_exists("$OE_SITE_DIR/images/practice_logo.jpeg")) $practice_logo =  "$OE_SITE_DIR/images/practice_logo.jpeg";
+  else $practice_logo = '';
+   $dir = explode('/', $practice_logo) ;
+   $practice_logo = "$dir[4]/$dir[5]/$dir[6]/$dir[7]/$dir[8]" ;
+   if ($practice_logo) {
+        echo "<img src='../../../../$practice_logo' align='left' style='width : 56px ; height : 90.6px'><br />\n";
```
Comment (teryhill) [inline interface/patient_file/report/custom_report.php]: Could a variable be added to the globals like $OE_SITE_DIR to get this. The hard coding of the path information will cause issues down the road when things are changes and someone forgets to change this.

File: interface/patient_file/report/custom_report.php [hunk 4]
```diff
@@ -484,12 +484,17 @@ function next_prev(action){
...
-        echo "<img src='$practice_logo' align='left'><br />\n";
-     } 
+  //use logo if it exists in directory
+  if (file_exists("$OE_SITE_DIR/images/practice_logo.png"))  $practice_logo = "$OE_SITE_DIR/images/practice_logo.png";
+  else if (file_exists("$OE_SITE_DIR/images/practice_logo.gif"))  $practice_logo = "$OE_SITE_DIR/images/practice_logo.gif";
+  else if (file_exists("$OE_SITE_DIR/images/practice_logo.jpg"))  $practice_logo = "$OE_SITE_DIR/images/practice_logo.jpg";
+  else if (file_exists("$OE_SITE_DIR/images/practice_logo.jpeg")) $practice_logo =  "$OE_SITE_DIR/images/practice_logo.jpeg";
+  else $practice_logo = '';
+   $dir = explode('/', $practice_logo) ;
+   $practice_logo = "$dir[4]/$dir[5]/$dir[6]/$dir[7]/$dir[8]" ;
```
Comment (teryhill) [inline interface/patient_file/report/custom_report.php]: I am not sure what you are doing here>

File: interface/patient_file/report/custom_report.php (hunk 3 above)
Comment (aethelwulffe) [reply interface/patient_file/report/custom_report.php]: Should be a global config to use an optional file specifically for this purpose...not sure if we even need a logo on a report like this at all.
  My final vote is that either we make the effort to put in a letterhead feature, or get rid of the image here.  It does nothing for the quality of the report.

Comment (2208Abhinav) [reply interface/patient_file/report/custom_report.php]: I also suggest that there is no specific need of image in the report.

Comment (aethelwulffe) [reply interface/patient_file/report/custom_report.php]: Abhinav,
Could you just do us all a favor and get rid of the image completely for now?  That is the simple and sensible solution.  It should have never been there in the first place.
Thanks,
-Art

Comment (2208Abhinav) [reply interface/patient_file/report/custom_report.php]: @aethelwulffe I suggest two options 
1)To completely remove the image.
2)To put a general medical logo like a plus sign or something like that.
Which one you favor?

Comment (aethelwulffe) [reply interface/patient_file/report/custom_report.php]: Totally remove the logo is my preference, but I am open to either option.

Comment (2208Abhinav) [reply interface/patient_file/report/custom_report.php]: Thank you for telling I will remove it.

Comment (aethelwulffe) [reply interface/patient_file/report/custom_report.php]: Just waiting on the new commit, and then I will test and merge.  I imagine this has been a very irritating issue for you.  Sorry such a simple thing got so complex.

**Comment by 2208Abhinav** [comment]
I have completely removed the image, @aethelwulffe  please review the changes.
And sorry for the long absence, I got ill in between the process but now I am back to work :smiley:

**Comment by 2208Abhinav** [comment]
I also worked for the general medical  logo.
Here's the result
screenshot from 2018-01-27 19-20-49

**Comment by aethelwulffe** [comment]
Thanks!
  We can worry about adding any letterhead or logos in another project.  This is good, clean, tested, and approved...and a very simple commit.
Merging....

**Pull request merged**